### PR TITLE
API call helper functions added

### PR DESCRIPTION
**Pull request by Princy2909**
This code provides a collection of functions to interact with both public and private APIs using Axios. It supports various HTTP methods (GET, POST, PUT, PATCH, DELETE) and implements OAuth token authentication for accessing private endpoints. It facilitates seamless communication with APIs while ensuring secure authentication and data transmission.
I have tested this code by calling public APIs

### CODE REVIEW

**Comment by Jha2508** [comment]
Code looks good, some unified concerns for development: 
1. Fetch all string resources from constants
2. Fetch the url for the API URI from some config
3. Testing code should be kept separate , no need to push that to `App.tsx`. Make a new file `TestApi.tsx` , do testing in your component and for testing use the component in `App.tsx`

**Comment by Princy2909** [comment]
@Jha2508   Changes have been done

File: TestAPI.tsx [hunk 1]
```diff
@@ -0,0 +1,171 @@
+import { View, TouchableOpacity, Button, Modal, Text, StyleSheet } from 'react-native'
+import React, { useState } from 'react'
+import { getPublicAPI, getPrivateAPI, patchPrivateAPI, patchPublicAPI, deletePrivateAPI, deletePublicAPI } from './APICall'
+import { postPrivateAPI, postPublicAPI, putPrivateAPI, putPublicAPI } from './APICall'
+
+export default function TestAPI() {
```
Comment (Jha2508) [inline TestAPI.tsx:23]: docs

File: TestAPI.tsx [hunk 2]
```diff
@@ -0,0 +1,171 @@
...
+    const endpoint = 'albums/1';       //replace it with your endpoint
+    const userData = {
+        userId: 1,
+        id: 1,
+        title: "rehenderit"
+    };
+    const [modalVisible1, setModalVisible1] = useState(false);
+    const [modalVisible2, setModalVisible2] = useState(false);
+
+    const handlePublicButton = () => {
```
Comment (Jha2508) [inline TestAPI.tsx:45]: docs for all functions

File: App.tsx [hunk 3]
```diff
@@ -1,118 +1,11 @@
-/**
```
Comment (Jha2508) [inline App.tsx]: No changes should be there on `App.tsx` file at all. please revert the change to this file

**Review by Jha2508 — APPROVED** [review]

File: App.tsx (hunk 3 above)
Comment (Princy2909) [reply App.tsx]: Done
